clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -10835,6 +10835,7 @@ def err_sycl_kernel_incorrectly_named : Error<
   "kernel %select{name is missing"
   "|needs to have a globally-visible name"
   "|name is invalid. Unscoped enum requires fixed underlying type"
+  "|name cannot be or use std::nullptr_t"
```

Comment:
Just wondering, why we disallow only nullptr_t?
Your PR message says " sounds like we don't want to encourage using builtin types as kernel names. "

Reply:
nullptr_t is special because builtin type names are (typically) accessible without a header, but nullptr_t is a builtin type that isn't known by any name until we include a header (or forward declare). The "don't want to encourage using builtins" comes from [rolandschulz's comment](https://github.com/intel/llvm/pull/1579#pullrequestreview-406823625) -- "Adding extra support for std types is not needed," and, "there is no promise that std types work as lambda name." But I may have misinterpreted his comments, and others asking him for clarification.